src/django/pfb_analysis/models.py

```diff
@@ -142,26 +190,22 @@ def create_environment(**kwargs):
                                      containerOverrides=container_overrides)
         try:
             self.batch_job_id = response['jobId']
-            # TODO: Possibly refactor status to use the AWS Batch status rather than some
-            #       custom status we may not have the ability to easily update
-            self.status = self.Status.IMPORTING
             self.save()
         except (botocore.exceptions.BotoCoreError, KeyError):
             logger.exception('Error starting AnalysisJob {}'.format(self.uuid))
 
-    def batch_job_status(self):
-        """ Return current AWS Batch job status for this job
 
-        List of available statuses: http://docs.aws.amazon.com/batch/latest/userguide/jobs.html
-        TODO: Refactor to cache in db?
+class AnalysisJobStatusUpdate(models.Model):
+    job = models.ForeignKey(AnalysisJob, related_name='status_updates', on_delete=models.CASCADE,
+                            null=True, blank=True)
+    status = models.CharField(choices=AnalysisJob.Status.CHOICES, max_length=12)
+    step = models.CharField(max_length=50, null=True, blank=True)
+    message = models.CharField(max_length=256, null=True, blank=True)
+    timestamp = models.DateTimeField(default=datetime.utcnow)
 
-        """
-        if not self.batch_job_id:
-            return None
-        client = boto3.client('batch')
-        try:
-            jobs = client.describe_jobs(jobs=[self.batch_job_id])['jobs']
-            return jobs[0]['status']
-        except KeyError:
-            logger.exception('Error retrieving AWS Batch job status for job'.format(self.uuid))
-            return None
+    def save(self, *args, **kwargs):
```

Comment:
I think my thought was that if we end up using API calls the timing could be subject to enough variables that we might want to pass the timestamp with the request to keep it precise and consistent.  Hence the overridable default rather than an `auto_now_add`.  I didn't use `PFBModel` because 1) I do think they should be immutable so the `modified` fields wouldn't be useful and 2) `PFBModel`requires `user`, which would be redundant.